### PR TITLE
[huggingface_pytorch] Update buildspec to re-release pt-1.9.1

### PR DESCRIPTION
**Pull request by philschmid**
*GitHub Issue #, if available:*

Update `buildspec.yml` to the content of `buildspec-1-9-1.yml` to re-release 1.9.1 images to fix `sagemaker-inference` log issue. 

* https://github.com/aws/sagemaker-huggingface-inference-toolkit/issues/61

> the PR by default builds off of buildspec.yml. In order for you to build/test the 1.9.1 images in a PR, you need to make a temp commit to copy the contents of buildspec-1-9-1.yml into buildspec.yml

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by arjkesh** [comment]
@philschmid there are 10 tests currently failing.

Three of the failing tests are "test_repo_anaconda_not_present" - this one @kevinyang8 can help debug. Since the training image is built on top of a released DLC, won't be able to fix the "training" image in this PR until the base DLC is re-released, so we should prioritize updating the PT 1.9 dockerfiles as well

**Comment by kevinyang8** [comment]
@philschmid To fix your "test_repo_anaconda_present" tests, you can merge [this](https://github.com/aws/deep-learning-containers/pull/1880) PR into your PR which makes the necessary changes to pass the test. We are currently prioritizing updating the PT 1.9 images with similar changes and releasing it. You can track the progress of that via [this](https://github.com/aws/deep-learning-containers/pull/1877) PR.

File: huggingface/pytorch/buildspec.yml [hunk 1]
```diff
@@ -29,21 +102,21 @@ context:
...
-    tag_python_version: &TAG_PYTHON_VERSION py38
-    cuda_version: &CUDA_VERSION cu113
-    os_version: &OS_VERSION ubuntu20.04
-    transformers_version: &TRANSFORMERS_VERSION 4.17.0
-    datasets_version: &DATASETS_VERSION 1.18.4
-    tag: !join [ *VERSION, '-', 'transformers', *TRANSFORMERS_VERSION, '-', *DEVICE_TYPE, '-', *TAG_PYTHON_VERSION, '-',
-                 *CUDA_VERSION, '-', *OS_VERSION ]
-    docker_file: !join [ docker/, *SHORT_VERSION, /, *DOCKER_PYTHON_VERSION, /,
-                         *CUDA_VERSION, /Dockerfile., *DEVICE_TYPE ]
+  # BuildHuggingFacePytorchGpuPy37Cu111TrainingDockerImage:
```
Comment (shantanutrip) [inline huggingface/pytorch/buildspec.yml:105]: I see that `BuildHuggingFacePytorchGpuPy37Cu111TrainingDockerImage` has Py37 in it, can we change it to Py38 like: `BuildHuggingFacePytorchGpuPy38Cu111TrainingDockerImage`

Comment (philschmid) [reply huggingface/pytorch/buildspec.yml:105]: The Training Image won't be re-released. So it will stay commented out. The only reason for this PR is that we have to re-release the 1.9.1 DLCs since there was a bug in the `sagemaker-infernece` package. @arjkesh told me back then that we cannot re-release a specific `buildspec-xxx.yaml` so i should open a PR and adjust the `buildspec.yaml` to only include the changes to then afterwards revert it to the correct one. If that's wrong please let me know

**Comment by tejaschumbalkar** [comment]
@philschmid Can you close the PR if it is not valid or move it to draft if it is not being worked currently?
